gap/SemilinearMatrixGroups.gi

```diff
@@ -137,11 +137,86 @@ function(n, q, s)
     return MatrixGroupWithSize(F, Concatenation(AandB, [C, D]), size);
 end);
 
+# For a sesquilinear form beta on the vector space GF(q ^ s) ^ (d / s) given 
+# by the Gram matrix <form>, we can interpret any u, v in GF(q) ^ d as elements of 
+# GF(q ^ s) ^ (d / s) via the standard basis. Taking the trace of beta(u, v)
+# yields a sesquilinear form on the vector space GF(q) ^ d.
+#
+# Takes a unitary form and returns a unitary form if <type> = "U-U"; takes a
+# unitary form and returns a symplectic form if <type> = "U-S"; takes a
+# symplectic form and returns a symplectic form if <type> = "S-S". 
+BindGlobal("TakeTraceOfSesquilinearForm",
```

Comment:
Is this really a good name? Perhaps something like `ReinterpretSesquilinearForm` or `LiftSesquilinearForm` would be more appropriate?

Reply:
Well, I'd say it is definitely not lifting since we take a sesquilinear form on a vector space over a bigger field `GF(q ^ s)` and turn it into a sesquilinear form on a vector space over a smaller field `GF(q)`.

I thought the name was really quite fitting since if `f` is a sesquilinear form on `GF(q ^ s) ^ (d / s)`, then the function returns a sesquilinear form `f'` on `GF(q) ^ d` satisfying `f'(u, v) = Tr(f(u, v))`, i.e. one literally takes the trace of the value of the sesquilinear form `f` to get the value of `f'`.

Reply:
Well, it doesn't return `Tr(f(u,v))` though, it returns `Tr\circ f`, so that would suggest `ComposeTraceAndForm` would fit that. But even if we stay with your argument: then why "Take" and not just "TraceOfSesquilinearForm" ???

Anyway, this is not important at all for the PR overall, and can be changed later, if at all.